sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -136,20 +137,45 @@ func (k *listenerImpl) handleAllEvents() {
 	handle(k.context, roleInformer, dispatchers.ForRBAC(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
 	handle(k.context, clusterRoleInformer, dispatchers.ForRBAC(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
 
+	// For openshift clusters only
 	var osConfigFactory osConfigExtVersions.SharedInformerFactory
 	if k.client.OpenshiftConfig() != nil {
-		if ok, err := clusterOperatorCRDExists(k.client); err != nil {
-			log.Errorf("Error checking for cluster operator CRD: %v", err)
-		} else if !ok {
-			log.Warnf("Skipping cluster operator discovery....")
+		if resourceList, err := serverResourcesForGroup(k.client, osConfigGroupVersion); err != nil {
+			log.Errorf("Error checking API resources for group %q: %v", osConfigGroupVersion, err)
 		} else {
 			osConfigFactory = osConfigExtVersions.NewSharedInformerFactory(k.client.OpenshiftConfig(), noResyncPeriod)
+
+			if resourceExists(resourceList, osClusterOperatorsResourceName) {
+				log.Infof("Initializing %q informer", osClusterOperatorsResourceName)
+				handle(k.context, osConfigFactory.Config().V1().ClusterOperators().Informer(), dispatchers.ForClusterOperators(), k.outputQueue, nil, noDependencyWaitGroup, stopSignal, &eventLock)
+			}
+
+			if env.RegistryMirroringEnabled.BooleanSetting() {
+				if resourceExists(resourceList, osImageDigestMirrorSetsResourceName) {
+					log.Infof("Initializing %q informer", osImageDigestMirrorSetsResourceName)
+					handle(k.context, osConfigFactory.Config().V1().ImageDigestMirrorSets().Informer(), dispatchers.ForRegistryMirrors(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
+				}
+
+				if resourceExists(resourceList, osImageTagMirrorSetsResourceName) {
+					log.Infof("Initializing %q informer", osImageTagMirrorSetsResourceName)
+					handle(k.context, osConfigFactory.Config().V1().ImageTagMirrorSets().Informer(), dispatchers.ForRegistryMirrors(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
+				}
+			}
 		}
 	}
-	// For openshift clusters only
-	if osConfigFactory != nil {
-		handle(k.context, osConfigFactory.Config().V1().ClusterOperators().Informer(), dispatchers.ForClusterOperators(),
-			k.outputQueue, nil, noDependencyWaitGroup, stopSignal, &eventLock)
+
+	var osOperatorFactory osOperatorExtVersions.SharedInformerFactory
+	if k.client.OpenshiftOperator() != nil && env.RegistryMirroringEnabled.BooleanSetting() {
```

Comment:
can you explain why we have these conditions?

Reply:
Assuming referring to `k.client.OpenshiftOperator() != nil`? It's possible for the various OpenShift* clients to fail initialization (see [api ref](https://pkg.go.dev/github.com/openshift/client-go@v0.0.0-20230718165156-6014fb98e86a/operator/clientset/versioned#NewForConfig)) - in which case sensor would panic if this check isn't there.

Reply:
oh never mind I thought this was the same condition as before `if k.client.OpenshiftConfig() != nil`

Reply:
what is the difference between the two anyway?

Reply:
I don't know the justification without digging into `client-go` sources, but each API 'group' has a unique interface required to produce the associated informer factory.  `ImageContentSourcePolicy` is part of the `Operator` group which has a different interface from `ImageDigestMirrorSet` which is part of the `Config` group, and so on.  Is API list w/ groups [here](https://docs.openshift.com/container-platform/4.13/rest_api/index.html)